wlroots/wlr_types/presentation_time.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) 2022 Aakash Sen Sharma
+
+from pywayland.server import Display, Signal
+from wlroots import PtrHasData, ffi, lib
+from wlroots.backend import Backend
+from wlroots.wlr_types import Output, Surface
+
+
+class PresentationEvent(PtrHasData):
+    def __init__(self, ptr) -> None:
+        self._ptr = ptr
+
+    def from_output(self, output_event: Signal):
+        lib.wlr_presentation_event_from_output(self._ptr, output_event._ptr)
+
+    @property
+    def output(self) -> Output:
+        return Output(self._ptr.output)
```

Comment:
From what I can tell in the wlroots code, this pointer can be null, so that might need to be checked here

Reply:
```patch
diff --git a/wlroots/wlr_types/presentation_time.py b/wlroots/wlr_types/presentation_time.py
index 54b74d0..316919c 100644
--- a/wlroots/wlr_types/presentation_time.py
+++ b/wlroots/wlr_types/presentation_time.py
@@ -15,6 +15,8 @@ class PresentationEvent(PtrHasData):
 
     @property
     def output(self) -> Output:
+        if self._ptr.output == ffi.NULL:
+            return None
         return Output(self._ptr.output)
 
     @property
```
Is this alright?

Reply:
Yep, though you'll also need to update the return type annotation to `Output | None`.